cpp/oneapi/dal/backend/primitives/reduction/reduction_rm_cw_dpc.cpp

```diff
@@ -0,0 +1,413 @@
+    const UnaryOp& unary,
+    const event_vector& deps) const {
+    return this->operator()(input, output, width, height, width, binary, unary, deps);
+}
+
+template <typename Float, typename BinaryOp, typename UnaryOp>
+sycl::nd_range<2> reduction_rm_cw_inplace<Float, BinaryOp, UnaryOp>::get_range(
+    const std::int64_t width) const {
+    const sycl::range<2> local(1, wg);
+    const sycl::range<2> global(width, wg);
+    return sycl::nd_range<2>(global, local);
+}
+
+template <typename Float, typename BinaryOp, typename UnaryOp>
+typename reduction_rm_cw_inplace<Float, BinaryOp, UnaryOp>::kernel_t
+reduction_rm_cw_inplace<Float, BinaryOp, UnaryOp>::get_kernel(const Float* input,
+                                                              Float* output,
+                                                              const std::int64_t height,
+                                                              const std::int64_t stride,
+                                                              const BinaryOp& binary,
+                                                              const UnaryOp& unary) {
+    return kernel_t(input, output, height, stride, binary, unary);
+}
+
+#define INSTANTIATE(F, B, U) template class reduction_rm_cw_inplace<F, B, U>;
+
+#define INSTANTIATE_FLOAT(B, U)                \
+    INSTANTIATE(double, B<double>, U<double>); \
+    INSTANTIATE(float, B<float>, U<float>);
+
+INSTANTIATE_FLOAT(min, identity)
+INSTANTIATE_FLOAT(min, abs)
+INSTANTIATE_FLOAT(min, square)
+
+INSTANTIATE_FLOAT(max, identity)
+INSTANTIATE_FLOAT(max, abs)
+INSTANTIATE_FLOAT(max, square)
+
+INSTANTIATE_FLOAT(sum, identity)
+INSTANTIATE_FLOAT(sum, abs)
+INSTANTIATE_FLOAT(sum, square)
+
+#undef INSTANTIATE_FLOAT
+
+#undef INSTANTIATE
+
+template <typename Float, typename BinaryOp, typename UnaryOp>
+class kernel_reduction_rm_cw_inplace_local {
+    using acc_t =
+        sycl::accessor<Float, 1, sycl::access::mode::read_write, sycl::access::target::local>;
+    using inp_t = const Float*;
+    using out_t = Float*;
+
+public:
+    kernel_reduction_rm_cw_inplace_local(acc_t cache_,
+                                         inp_t const input_,
+                                         out_t const output_,
+                                         const std::int64_t height_,
+                                         const std::int32_t lstride_,
+                                         const BinaryOp& binary_,
+                                         const UnaryOp& unary_)
+            : cache{ cache_ },
+              input{ input_ },
+              output{ output_ },
+              unary{ unary_ },
+              binary{ binary_ },
+              height{ height_ },
+              lstride{ lstride_ } {}
+
+    void operator()(sycl::nd_item<2> it) const {
+        using sycl::ONEAPI::reduce;
+        // Common for whole WG
+        const auto col_idx = it.get_global_id(0);
+        const auto loc_idx = it.get_global_id(1);
```

Comment:
It would be better to user get_local_id(1). It might be the same for the particular case but eventually become wrong with ndrange change in a calling function.

Reply:
Fixed